main.go

```diff
@@ -127,6 +127,9 @@ func (s *slackOutput) notificationServiceRoutes(fields map[string]interface{}) [
 		return []decode.NotificationRoute{}
 	}
 
+	// Convert value into a string version, if it's not, for consistency.
+	alert.Value = fmt.Sprintf("%v", alert.Value)
```

Comment:
Would you mind adding a unit test to verify this works as expected? 

golang's sprintf command often prints out some pretty ugly stuff.

Here's how I handled something similar in the alerts consumer:

- type switch: https://github.com/Clever/kinesis-alerts-consumer/blob/master/alerts_consumer.go#L95-L110
- unit test: https://github.com/Clever/kinesis-alerts-consumer/blob/master/alerts_consumer_test.go#L103-L155

Reply:
`%v	the value in a default format`
which is pretty much exactly what I want to convert a (random) interface into a string.
But if you'd prefer the type switch, I can do that instead.

But sure, I can add a test.

Reply:
Cool - seems like works well! I don't know where my bias comes from, perhaps I've made mistakes in the past that resulted in format errors

```
Wrong type or unknown verb: %!verb(type=value)
	Printf("%d", hi):          %!d(string=hi)
Too many arguments: %!(EXTRA type=value)
	Printf("hi", "guys"):      hi%!(EXTRA string=guys)
Too few arguments: %!verb(MISSING)
	Printf("hi%d"):            hi%!d(MISSING)
Non-int for width or precision: %!(BADWIDTH) or %!(BADPREC)
	Printf("%*s", 4.5, "hi"):  %!(BADWIDTH)hi
	Printf("%.*s", 4.5, "hi"): %!(BADPREC)hi
Invalid or invalid use of argument index: %!(BADINDEX)
	Printf("%*[2]d", 7):       %!d(BADINDEX)
	Printf("%.[2]d", 7):       %!d(BADINDEX)
```

https://golang.org/pkg/fmt/

Anyway, seems good at present!

Test wouldn't hurt, ofc

Reply:
Yeah, printf with non-%v format strings is terrible, because if the type doesn't match, go won't cast it, and you get weird errors like that. %v is the workaround (IMO) to make it "just work".

However, you were absolutely right to push a test, as (a) I had the override one stanza too low (so it wasn't changing the output) and (b) I needed a nil check first.

Want to take another look?